### PR TITLE
Validate operator before attempting to update number filter

### PR DESCRIPTION
**Pull request by anguss00**
Fixes issue #1744

### CODE REVIEW

**Comment by dnahodil** [comment]
I'm just trying this out and it looks good. However, I'm just thinking it might be good to set the drop-down menu to be invalid to show the user that they should change that so their filter is applied. What do you think about that? I know this wasn't listed in 'what should happen'.

**Comment by anguss00** [comment]
Yes, that's definitely a good point. I was thinking about doing that actually before I pushed this up... Can definitely implement if you think it's worthwhile @dnahodil

**Comment by anguss00** [comment]
So have a red line around the numberfield while the operator dropdown is not set?

**Comment by dnahodil** [comment]
Yeah, let's do it. I would make the dropdown invalid, actually. The number is fine in this case, the problem is that the drop-down is not set.

**Comment by dnahodil** [comment]
Hmm. We don't actually do the opposite. If you select and operator but haven't yet enterred a number we don't highlight the number field. We should probably do that, too. :)

**Comment by anguss00** [comment]
Good point, I'm on it

**Comment by pmbohm** [comment]
@anguss00 Ive added what I think is good validation to the NumberFilterPanel, and now i read your discussion with @dnahodil. I wont look too close at what was said in case it conflicts with my changes :)

**Comment by anguss00** [comment]
Haha thanks @pmbohm I'll have a look now

**Comment by pmbohm** [comment]
@anguss00 the updating of the map when the user selects none is now done. Let me know if this is good and then i will squash yours an my work to one. The differences are large and confusing to leave two commits i think?

File: web-app/js/portal/filter/ui/NumberFilterPanel.js [hunk 1]
```diff
@@ -115,10 +154,8 @@ Portal.filter.ui.NumberFilterPanel = Ext.extend(Portal.filter.ui.BaseFilterPanel
         });
 
         this._fireAddEvent();
+        trackFiltersUsage('filtersTrackingNumberAction', this._getTrackUsageLabel(), this.layer.name);
```
Comment (anguss00) [inline web-app/js/portal/filter/ui/NumberFilterPanel.js]: Only issue is here; if I make an operator selection and then add a value in the first field, trackUsage is called twice (once for the operator and one for the numberField selection).

**Comment by anguss00** [comment]
Looks good aside from one comment @pmbohm

**Comment by pmbohm** [comment]
@anguss00 trackchanges hole now plugged. Commits squashed

**Comment by anguss00** [comment]
Me gusta

**Pull request merged**